kymatio/backend/torch_backend.py

```diff
@@ -1,230 +1,242 @@
+    
+            ctx.save_for_backward(x, output)
+    
+            return output
+    
+        @staticmethod
+        def backward(ctx, grad_output):
+            """Backward pass of the modulus
+    
+            This is a static method which does not require an instantiation of the
+            class.
+    
+            Arguments
+            ---------
+            ctx : context object
+                Collected during the forward pass. These are automatically added
+                by PyTorch and should not be touched. They are then used for the
+                backward pass.
+            grad_output : tensor
+                The gradient with respect to the output tensor computed at the
+                forward pass.
+    
+            Returns
+            -------
+            grad_input : tensor
+                The gradient with respect to the input.
+            """
+            x, output = ctx.saved_tensors
+            if ctx.dim is not None and ctx.keepdim is False and x.dim() != 1:
+                grad_output = grad_output.unsqueeze(ctx.dim)
+                output = output.unsqueeze(ctx.dim)
+    
+            grad_input = x.mul(grad_output).div(output)
+    
+            # Special case at 0 where we return a subgradient containing 0
+            grad_input.masked_fill_(output == 0, 0)
+    
+            return grad_input
+    
+    # shortcut for ModulusStable.apply
+    modulus = ModulusStable.apply
+    
+    class Modulus():
+        """This class implements a modulus transform for complex numbers.
+    
+            Usage
+            -----
+            modulus = Modulus()
+            x_mod = modulus(x)
+    
+            Parameters
+            ---------
+            x : tensor
+                Complex torch tensor.
+    
+            Returns
+            -------
+            output : tensor
+                A tensor with the same dimensions as x, such that output[..., 0]
+                contains the complex modulus of x, while output[..., 1] = 0.
+        """
+        def __call__(self, x):
+            complex_contiguous_check(x)
+            norm = modulus(x)[..., None]
+            return norm
+    
+    def complex_contiguous_check(self, x):
+        self.complex_check(x)
+        self.contiguous_check(x)
     
-    complex_contiguous_check(A)
-
-    if A.shape[-len(B.shape):-1] != B.shape[:-1]:
-        raise RuntimeError('The filters are not compatible for multiplication.')
 
-    if A.dtype is not B.dtype:
-        raise TypeError('Input and filter must be of the same dtype.')
+    def contiguous_check(self, x):
+        if not x.is_contiguous():
+            raise RuntimeError('Tensors must be contiguous.')
+    
 
-    if B.device.type == 'cuda':
-        if A.device.type == 'cuda':
-            if A.device.index != B.device.index:
-                raise TypeError('Input and filter must be on the same GPU.')
+    def cdgmm(self, A, B, inplace=False):
```

Comment:
that was another PR that hasn't been merged yet.

Reply:
Actually you are right.

Reply:
I based this PR based off of the 1d test pr. The 1d PR had not been rebased ontop of the change. How do we want to proceed?

Reply:
such that the end result is that there is no more `inplace` stuff. I would suggest removing it here. Otherwise it might end up showing up again when this gets merged